### ISSUE TITLE
using the mysql monitor from another recipe breaks

### ISSUE THREAD

**Issue by joshsmoore** [DataDog/chef-datadog#230]
Hi I am trying to us the mysql monitor from another recipe.  I am calling it like this: 

````ruby
datadog_monitor 'mysql' do
  instances [
    {
      'server' => 'localhost',
      'user' => 'datadog',
      'pass' => datadog_password,
      'tags' => [
        "#{node['application_name']}-#{node.chef_environment}-db"
      ]
    }
  ]
  only_if do
    node['database-type'] == 'Mysql'
  end
end
````

However, when I run this I get this error in chef:

```
  * datadog_monitor[mysql] action add (up to date)
  * template[/etc/dd-agent/conf.d/mysql.yaml] action create

    ================================================================================
    Error executing action `create` on resource 'template[/etc/dd-agent/conf.d/mysql.yaml]'
    ================================================================================

    Chef::Exceptions::FileNotFound
    ------------------------------
    Cookbook 'db_config' (0.3.0) does not contain a file at any of these locations:
      templates/ubuntu-14.04/mysql.yaml.erb
      templates/ubuntu/mysql.yaml.erb
      templates/default/mysql.yaml.erb

    This cookbook _does_ contain: ['mysql_config.erb','mysql_database.yml.erb','postgres_database.yml.erb']

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/datadog/providers/monitor.rb

      9:   template "/etc/dd-agent/conf.d/#{new_resource.name}.yaml" do
     10:     owner 'dd-agent'
     11:     mode 00600
     12:     variables(
     13:       :init_config => new_resource.init_config,
     14:       :instances   => new_resource.instances
     15:     )
     16:     notifies :restart, 'service[datadog-agent]', :delayed
     17:   end
     18:   new_resource.updated_by_last_action(false)

    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/datadog/providers/monitor.rb:9:in `block in class_from_file'

    template("/etc/dd-agent/conf.d/mysql.yaml") do
      provider Chef::Provider::Template
      action "create"
      retries 0
      retry_delay 2
      guard_interpreter :default
      path "/etc/dd-agent/conf.d/mysql.yaml"
      backup 5
      atomic_update true
      source "mysql.yaml.erb"
      variables {:init_config=>{}, :instances=>[{"server"=>"localhost", "username"=>"datadog", "password"=>"<removed>", "tag"=>["resumecompanion-rc_production-db"]}]}
      cookbook_name "db_config"
      mode 384
      owner "dd-agent"
    end
````

I have fixed this by changing the `providers/monigor.rb` to include `cookbook 'data dog'` but I am wondering is there a better way to do this?

**Comment by miketheman** [comment]
Hi @joshsmoore,

This is due to the source cookbook not being a parameter that can be passed. The way this was designed was to not use the resource directly, rather to include the recipe in your run list to get the desired behavior.
See http://docs.datadoghq.com/guides/chef/#customizations for more examples.

We may be adding this in the next cookbook release, see #212.

**Comment by miketheman** [comment]
Closed via #212.
